### PR TITLE
DevTools components integration

### PR DESCRIPTION
**Pull request by vince19972**
Todos before merge:
- [x] Migrating components from DevTools UI to here
- [x] Organize stylesheets and `vuex` store
- [ ] Implement ESLint
- [ ] Organize directory structure

Next steps:
- Integrate Scaffold UI and DevTools UI
- Release Starport DevTools UI using this component library
- Create Starport typography and color system
- Revamp Scaffold UI

### CODE REVIEW

**Comment by vince19972** [comment]
Hey @fadeev , when you have time, can you please review the migration from DevTools, making sure it's in a good shape? If so, before we merge, I'll work on organizing the file structure, naming, and coding style. 

Currently I have a local branch of our DevTools UI linked to this repo with `npm link`, and it's all working smoothly. Once this PR is merged, we can then release DevTools UI that uses this extracted components. 

Let me know if this sounds like a good plan!

**Comment by fadeev** [comment]
Awesome! Checking it out!

**Comment by fadeev** [comment]
First things first, let's sort out the directory structure and organization. I believe at the top level (`@/components/`), we should have components users potentially would like to import. Right now we have one of those: `SpBlockExplorer`.  Since this is a "macro" component with a bunch going on, we can have a directory for it.

```
SpBlockExplorer/ <-- inside are components SpBlockExplorer.vue depends on
SpButton.vue
SpWhatever.vue
SpBlockExplorer.vue
```

This way we will hide the internal workings and a user of the lib will be able to see a list of exactly what we offer.

**Comment by fadeev** [comment]
To start using the lib, the only thing a user has to do is to dispatch a Vuex action and import a module. Let's try keeping it that way. Define what `SpBlockExplorer` component needs in terms of init logic and how the data should be structured. We can import non-namespaced modules into `cosmos.js` to split a large Vuex file, but let's try keeping it in one for now.

Ideally, user-facing components should not require any props, unless absolutely necessary. I don't believe a block explorer needs anything, so let's try to keep it vanilla: I should be able to use it like `<sp-block-exlporer />`.

I know this is a lot, but I'm afraid if we merge it as is we'll have a hard time organizing it further down the road.

**Comment by fadeev** [comment]
There is also a bit of an issue with block explorer going all classic mac os sad face on me, but we can deal with it later (the problem is with too much data within Vuex).

Screenshot 2020-10-28 at 12 22 18

**Comment by vince19972** [comment]
@fadeev No problem! I'll work on the directory structure and store integration today. 

Seeing the sad face is interesting though, because I've never seen it haha. I'll run Starport for a longer time again today. Let's see if I can meet the sad face. I've tried running it until 5000 something blocks before, but it worked well, btw.

**Comment by fadeev** [comment]
Yeah, no, let's take this one step at a time.

* `components` directory should have one file or directory per exported component.
* Each "helper" is used only once, they should not have their own top-level directory.
* More than 500 lines of CSS is globally imported. This affects everything in the app. And they include common non-prefixed classes, such as `.fade-enter-active`.
* Vuex actions are public API. Slicing and dicing them willy nilly is not good.

In short, before we can merge anything from here, make sure there **exactly two** new directories in components: `SpBlockInfoCard/` and `SpBlockExplorer/`, and one new file in `store`. No helpers, no layouts, and no global styles.

**Comment by vince19972** [comment]
Thanks for the feedback. Here are my thoughts:

- I agree about the structure in `components` directory. But we're not clear about what files or components we want to expose yet, due to the lack of a clear system. For example, the icons vue files are in the directory, but they're not exported either. So maybe we should have another directory to organize the atomic pieces for building up the exposed components. 
- I don't understand why helper/utility functions are not supposed to be organized into files according to their individual purposes. These are functions without side effects that are very likely to be, and some of them already are, used in various occasions. If we don't extract them away, where are we keeping them?
- You're right, the `fade` classes should be prefixed. The `main.css` file is optional to include though. I believe global style variables are required to build up our design system, including colors, typography and more. And the point of using a component library for a developer should be enjoying the convenience of the entire bundle of predefined styles we've prepared, right? Of course, from the point of developing this component library, we can import the individual `colors.css` or `typography.css` files repeatedly in each `vue` file, though I don't see the reason. 
- Each module in the store directory is responsible for a particular purpose. I'm just not sure about the advantage of squashing them altogether instead of separating them. In most scenarios, users only need to import the `@tendermint/vue/src/store/cosmos.js` file once, and execute the `init` function once, without dealing with other actions. In general, I see the two advantages of organizing the store into smaller modules: 1) easier for us to maintain as the library grows bigger, and 2) easier for developers who'd like to dig into the library for customizing it, or using the vuex actions.

**Comment by vince19972** [comment]
Hey @fadeev, let me know if the latest commits meet the temporary requirements! 

I suppose we don't need to merge this into `develop` branch right now, if we want to restructure `vuex` into a more decent shape; instead, I can create PRs to merge into `ui/develop` branch (or a better named branch), and merge it into `develop` branch as it gets done.

**Review by fadeev — APPROVED** [review]
Yes, this is good, thank you. We can indeed keep this branch open to restructure Vuex. I suggest creating an issue for that to describe proposed changes.
